eth/eventdatahandler/event_data_handler.go

```diff
@@ -276,22 +250,162 @@ func (edh *EventDataHandler) processEvent(txn eventdb.RW, event ethtypes.Log) (T
 			return nil, fmt.Errorf("handle FeeRecipientAddressUpdated: %w", err)
 		}
 
-		task := func() error {
-			if !updated {
-				return nil
-			}
+		if !updated {
+			return nil, fmt.Errorf("provided recipient address is the same")
```

Comment:
An error returned from handler means that we stop handling events. However, we don't need to stop if handle fee recipient address was not updated. We just don't need to create a task in this case. Let's keep the original behavior